File: Python/longest-word-in-dictionary.py
class Solution:
    def longestWord(self, words):
        """
        :type words: List[str]
        :rtype: str
        """
        words.sort()
        s = set([''])
        res = ''

        for word in words:
            if word[:-1] in s:
                s.add(word)
                if len(word) > len(res):
                    res = word

        return res

File: Swift/longest-word-in-dictionary.swift
class Solution {
    func longestWord(_ words: [String]) -> String {
        let sortedWords = words.sorted()
        var set = Set([""])
        var res = ""

        print(sortedWords)

        for word in sortedWords {
            if set.contains(String(word.prefix(word.count-1))) {
                set.insert(word)
                if word.count > res.count {
                    res = word
                }
            }
        }

        return res
    }
}
